### DISCUSSION TITLE
Disabling sharing options

### DISCUSSION THREAD

**Discussion by madduck** [requarks/wiki#6628] [Help / Questions] [answered]
Apart from the fact, that use of the `mdi-sharing-variant` icon (rather than `mdi-sharing`) causes there to be no icon in my setup, I'd like to also know if it's possible to add/remove items from that list:
![image](https://github.com/requarks/wiki/assets/195073/550972a3-452e-4e37-86f7-999b2b1ba763)
For instance, I want to remove Facebook&Twitter as options. Is this possible?

Thanks,
martin

**Comment by NGPixel** [comment, marked as answer]
No.

In v3, you'll be able to disable that menu altogether and the options will be Copy URL and Email only, as the rest is pointless for the vast majority of users.

**Comment by madduck** [comment]
Oh okay, thanks for letting us know!

**Comment by bobapplemac** [comment]
In the meantime, for v2, you can add the following script block to the Head HTML Injection area on the admin page

`<script>
(function waitForBodyThenObserve() {
  if (!document.body) {
    return setTimeout(waitForBodyThenObserve, 50);
  }

  const labelsToHide = new Set([
    "Facebook",
    "LinkedIn",
    "Reddit",
    "Telegram",
    "Twitter",
    "Viber",
    "Weibo",
    "Whatsapp",
    // Add more labels as needed
  ]);

  const hideMenuItems = () => {
    let found = false;
    document.querySelectorAll('.v-list-item').forEach(item => {
      const label = item.querySelector('.v-list-item__title')?.textContent?.trim();
      if (labelsToHide.has(label)) {
        item.style.display = 'none';
        found = true;
      }
    });
    return found;
  };

  const observer = new MutationObserver(() => {
    if (hideMenuItems()) {
      observer.disconnect(); // One and done
    }
  });

  observer.observe(document.body, {
    childList: true,
    subtree: true,
  });
})();
</script>
`
